recipes/grass/build.sh

```diff
@@ -0,0 +1,145 @@
+
+#!/usr/bin/env bash
+# Allow build to continue even if some tests fail
+set -uxo pipefail
+
+# Extract version components from PKG_VERSION (set by conda-build from meta.yaml)
+# PKG_VERSION format: MAJOR.MINOR.PATCH (e.g., "8.4.1")
+IFS='.' read -r GRASS_MAJOR GRASS_MINOR GRASS_PATCH <<< "${PKG_VERSION}"
+GRASS_VERSION_DIR="grass${GRASS_MAJOR}${GRASS_MINOR}"
+echo "=== GRASS Version: ${PKG_VERSION} (using directory: ${GRASS_VERSION_DIR}) ===" >&2
+echo "=== Version components: MAJOR=${GRASS_MAJOR}, MINOR=${GRASS_MINOR}, PATCH=${GRASS_PATCH} ===" >&2
+
+# Respect conda build environment
+export CC=${CC:-${GCC:-gcc}}
+export CXX=${CXX:-${GXX:-g++}}
+export CFLAGS="${CFLAGS:-} -O2 -std=gnu99"
+export CXXFLAGS="${CXXFLAGS:-} -O2"
+export LDFLAGS="${LDFLAGS:-}"
+
+# Platform-specific settings
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    export LIBS="${LIBS:-} -liconv"
+    # macOS linker flags
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,${PREFIX}/lib"
+else
+    export LIBS="${LIBS:-} -liconv"
+fi
+
+# Ensure pkg-config finds things in $PREFIX
+export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig:${PREFIX}/share/pkgconfig:${PKG_CONFIG_PATH:-}"
+export LD_LIBRARY_PATH="${PREFIX}/lib:${LD_LIBRARY_PATH:-}"
+
+# Configure flags adapted from Dockerfile. Use conda prefixes and configs.
+CONFIG_FLAGS=(
+  --prefix="${PREFIX}"
+  --with-cxx
+  --enable-largefile
+  --without-opengl
+  --with-proj-share="${PREFIX}/share/proj"
+  --with-gdal="${PREFIX}/bin/gdal-config"
+  --with-geos
+  --with-sqlite
+  --with-cairo
+  --with-freetype "--with-freetype-includes=${PREFIX}/include/freetype2"
+  --with-fftw
+  --with-postgres "--with-postgres-includes=${PREFIX}/include"
+  --with-netcdf
+  --with-zstd
+  --with-bzlib
+  --with-pdal
+  --without-mysql
+  --with-blas
+  --with-lapack
+  --with-readline
+  --with-openmp
+)
+
+# Add X11 only on Linux
+if [[ "$OSTYPE" != "darwin"* ]]; then
+    CONFIG_FLAGS+=(--with-x)
+else
+    CONFIG_FLAGS+=(--without-x)
+fi
+
+# Use system python from conda env
+export PYTHON=python
+
+# Clean and build
+echo "=== Starting GRASS build ===" >&2
+make distclean || true
+echo "=== Running configure ===" >&2
+./configure ${CONFIG_FLAGS[@]} 2>&1 | tee /tmp/configure-output.txt
+echo "configure-complete" > /tmp/build-status.txt
+
+## IMPORTANT: Fix ICONVLIB immediately after configure and BEFORE make
```

Comment:
I don't know if it's exactly the same problem, but on FreeBSD (which macos shares tooling), we have some ["hacks"](https://github.com/freebsd/freebsd-ports/blob/main/Mk/Uses/iconv.mk#L47-L66) for iconv. afaik I can build grass (not in conda) with: `cmake ... CPPFLAGS+=-DLIBICONV_PLUG CFLAGS+=-DLIBICONV_PLUG CXXFLAGS+=-DLIBICONV_PLUG`